### DISCUSSION TITLE
Extending a collection

### DISCUSSION THREAD

**Discussion by JeffKrammer** [isar/isar#700] [Help] [answered]
Prior to the 3.0.0 release I had most of my collections extend a base collection. The base collection has several variables and methods that I would like to use for all the classes that extend it. With 3.0.0 I get this error when attempting to extend a collection.

Class must not have a supertype annotated with @Collection or @Embedded.

I'm looking for any recommendations or workarounds to achieve this same goal without adding each of those variables and methods to every collection class.

**Comment by simc** [comment, marked as answer]
You can use inheritance but don't annotate the base class with `@collection`
